### PR TITLE
Use OA template for CI

### LINKED ISSUE #178: Add a GitHub Action to automate the release process
There are a few GitHub Actions out there that automate the release process to PyPI.  For example, @StanczakDominik has been working on https://github.com/PlasmaPy/PlasmaPy/pull/1779, which I'm really excited about.  We could add a GitHub Action like that one to fiasco to facilitate quicker releases.

### PR DESCRIPTION
**Pull request by wtbarnes**
Fixes #178 

Simplify CI by using the OA template

### CODE REVIEW

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -6,72 +6,26 @@ on:
...
-          python-version: 3.9
-          toxenv: py39
-          cache_cmd: -- --dbase-download-dir ~/.chianti
-        - os: ubuntu-latest
+jobs:
+  test:
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@v1
+    with:
+      posargs: '--dbase-download-dir ~/.chianti'
+      toxdeps: "'tox<4' tox-pypi-filter"
```
Comment (Cadair) [inline .github/workflows/run-tests.yml:15]: You probs don't need this

File: .github/workflows/run-tests.yml [hunk 2]
```diff
@@ -6,72 +6,40 @@ on:
...
-        - os: ubuntu-latest
-          python-version: 3.9
-          toxenv: py39
-          cache_cmd: -- --dbase-download-dir ~/.chianti
-        - os: ubuntu-latest
+        - linux: pre-commit
+          posargs: ''
+      coverage: 'codecov'
+      cache-path: ~/.chianti
+      cache-key: chianti-${{ github.run_id }}
```
Comment (Cadair) [inline .github/workflows/run-tests.yml:31]: What scope is this caching over?

File: .github/workflows/run-tests.yml (hunk 1 above)
Comment (wtbarnes) [reply .github/workflows/run-tests.yml:15]: Oh I wasn't sure if this was fixed upstream/in the template

File: .github/workflows/run-tests.yml (hunk 2 above)
Comment (wtbarnes) [reply .github/workflows/run-tests.yml:26]: Every env can use the same database  so I don't want to download for every env.

My old cache key was this: https://github.com/wtbarnes/fiasco/blob/main/.github/workflows/run-tests.yml#L67. But I wasn't sure how to reproduce that here.

**Pull request merged**